miasm2/expression/simplifications_common.py

```diff
@@ -1049,9 +1064,18 @@ def simp_zeroext_eq_cst(expr_s, expr):
         return ExprInt(0, 1)
     return ExprOp(TOK_EQUAL, src, ExprInt(int(arg2), src.size))
 
-def simp_ext_eq_ext(expr_s, expr):
-    # A.zeroExt(X) == B.zeroExt(X) => A == B
-    # A.signExt(X) == B.signExt(X) => A == B
+def simp_cond_zeroext(_, expr):
+    """X.zeroExt()?(A:B) => X ? A:B"""
+    if not (expr.cond.is_op() and expr.cond.op.startswith("zeroExt")):
```

Comment:
This is also true for `signExt` (ie `X.signExt` != 0 <=> `X` != 0)

Reply:
done